test/unit/test-Lexer.js

```diff
@@ -0,0 +1,12 @@
+var assert    = require('assert');
+var process   = require('process');
+var test      = require('utest');
+
+// If we're on a Node runtime that should support ES6, run the ES6 tests.
+var nodeVersion = process.env.npm_config_node_version;
+if (/^[0-5][.]/.test(nodeVersion || '')) {
```

Comment:
TODO(mikesamuel): Is there a common place I can put this old-ES-engine-test-skipping machinery?